src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -260,4 +262,138 @@ PaymentSandbox::apply (PaymentSandbox& to)
+                    newBalance = oldBalance.zeroed();
+                    lowID = xrpAccount();
+                    highID = (*before)[sfAccount];
+                    break;
+                case ltRIPPLE_STATE:
+                    {
+                        lowID = (*before)[sfLowLimit].getIssuer();
+                        highID = (*before)[sfHighLimit].getIssuer();
+                        oldBalance = (*before)[sfBalance];
+                        highID = (*before)[sfAccount];
+                    }
+
+                    break;
+                case ltOFFER:
+                    // TBD
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (!before)
+        {
+            // insert
+            auto const at = after->getType ();
+            switch(at)
+            {
+                case ltACCOUNT_ROOT:
+                    newBalance = (*after)[sfBalance];
+                    oldBalance = newBalance.zeroed();
+                    lowID = xrpAccount();
+                    highID = (*after)[sfAccount];
+                    break;
+                case ltRIPPLE_STATE:
+                    {
+                        lowID = (*after)[sfLowLimit].getIssuer();
+                        highID = (*after)[sfHighLimit].getIssuer();
+                        newBalance = (*after)[sfBalance];
+                        oldBalance = newBalance.zeroed();
+                    }
+
+                    break;
+                case ltOFFER:
+                    // TBD
+                    break;
+                default:
+                    break;
+            }
+        }
+        else
+        {
+            // modify
+            auto const bt = before->getType ();
+            auto const at = after->getType ();
+            assert (at == bt);
+            switch(at)
+            {
+                case ltACCOUNT_ROOT:
+                    oldBalance = (*before)[sfBalance];
+                    newBalance = (*after)[sfBalance];
+                    lowID = xrpAccount();
+                    highID = (*after)[sfAccount];
+                    break;
+                case ltRIPPLE_STATE:
+                {
+                    lowID = (*after)[sfLowLimit].getIssuer();
+                    highID = (*after)[sfHighLimit].getIssuer();
+                    oldBalance = (*before)[sfBalance];
+                    newBalance = (*after)[sfBalance];
+                }
+
+                break;
+                case ltOFFER:
+                    // TBD
+                    break;
+                default:
+                    break;
+            }
+        }
+        // The following are now set, put them in the map
+        auto delta = newBalance - oldBalance;
+        auto const cur = newBalance.getCurrency();
+        result[std::make_tuple(lowID, highID, cur)] = delta;
+        auto r = result.emplace(std::make_tuple(lowID, lowID, cur), delta);
```

Comment:
Does the
```result[std::make_tuple(lowID, highID, cur)] = delta;```
duplicate the effort of the
```auto r = result.emplace(std::make_tuple(lowID, lowID, cur), delta);```?
I wonder if you could remove the line that array indexes into `result`.